### DISCUSSION TITLE
Wheel does not angle to the setpoint angle

### DISCUSSION THREAD

**Discussion by uwuscutely** [BroncBotz3481/YAGSL-Example#161] [Q&A]
I'm seeing this issue where the wheel will be a few degrees off from the setpoint if you switch from commanding full forward to full left/right with the control stick. However, when you release the control stick on the controller, it will reach the desired angle setpoint. Any ideas what could cause an issue like this? It's like there's a delay in the angle.

For further clarification, lets say I command full forward, the wheels will all face forward and drive forward. Let's say I switch to commanding full left/right without returning the control stick to neutral position, the wheels will turn to about 60 degrees instead of 90. Then when I let go of the control stick and it returns to neutral position, the wheels will go to 90 degrees. The controller is a Logitech controller.

**Comment by thenetworkgrinch** [comment]
When you go left/right from forward is it commanded to go left/right too? Are you using the correct conversion factors?

**Reply by uwuscutely** [reply]
Yeah the red indicator in FRC Web Components will show full left/right command, but the blue indicator is very clearly not aligned with it. I'm a mentor from a different team helping another team with their code so waiting on the full codebase to confirm the conversation factors. It's an MK4i running with NEOs and CANcoders.

**Reply by thenetworkgrinch** [reply]
The angle conversion factor should be `16.8` then.

Be sure that they used the right absolute encoder offsets too.

**Reply by uwuscutely** [reply]
I did make sure the absolute encoder does spin CCW+. Will confirm conversion faction when they get back to me. Side question, is there anything specific reason you would invert if everything is already CCW+?

**Reply by thenetworkgrinch** [reply]
It should all be CCW+ but teams frequently dont select the right modules when configuring it initially, sometimes inverting fixes it without moving modules around.

![image-48.png](https://github.com/BroncBotz3481/YAGSL-Example/assets/10247070/ee6a142a-a0ff-45c3-8536-57d71b34c337)

**Reply by uwuscutely** [reply]
If I'm understanding this image correctly. The module IDs itself were changed right?

**Reply by thenetworkgrinch** [reply]
Yes, you may have accidentally swapped front/back and left/right when initially inputting your module data.
